  @Test(expectedExceptions = ValidationException.class)
  public void shouldFailIfSecurityContextDiffersInPods() throws Exception {
    // given
    PodSpec podSpec1 =
        new PodSpecBuilder()
            .withSecurityContext(new PodSecurityContextBuilder().withRunAsUser(42L).build())
            .build();
    podSpec1.setAdditionalProperty("add1", 1L);
    PodData podData1 = new PodData(podSpec1, new ObjectMetaBuilder().build());

    PodSpec podSpec2 =
        new PodSpecBuilder()
            .withSecurityContext(new PodSecurityContextBuilder().withRunAsUser(43L).build())
            .build();
    podSpec2.setAdditionalProperty("add2", 2L);
    PodData podData2 = new PodData(podSpec2, new ObjectMetaBuilder().build());

    // when
    Deployment merged = podMerger.merge(Arrays.asList(podData1, podData2));

    // then
    // exception is thrown
  }

